pkg/nodeagent/apis/config/v1alpha1/types.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// NodeAgentBaseDir is the directory on the worker node that contains gardener-node-agent
+	// relevant files.
+	NodeAgentBaseDir = "/var/lib/gardener-node-agent"
+	// NodeAgentConfigPath is the file path on the worker node that contains the configuration
+	// of the gardener-node-agent.
+	NodeAgentConfigPath = NodeAgentBaseDir + "/configuration.yaml"
+	// NodeAgentInitScriptPath is the file path on the worker node that contains the init script
+	// of the gardener-node-agent.
+	NodeAgentInitScriptPath = NodeAgentBaseDir + "/gardener-node-init.sh"
+
+	// NodeInitUnitName is the name of the gardener-node-init systemd service.
+	NodeInitUnitName = "gardener-node-init.service"
+	// NodeAgentUnitName is the name of the gardener-node-agent systemd service.
+	NodeAgentUnitName = "gardener-node-agent.service"
+
+	// NodeAgentOSCSecretKey is the key inside the gardener-node-agent osc secret to access
+	// the encoded osc.
+	NodeAgentOSCSecretKey = "gardener-node-agent"
+	// NodeAgentOSCOldConfigPath is the file path on the worker node that contains the
+	// previous content of the osc
+	NodeAgentOSCOldConfigPath = NodeAgentBaseDir + "/previous-osc.yaml"
+
+	// NodeAgentTokenFilePath is the file path on the worker node that contains the shoot access
+	// token of the gardener-node-agent.
+	NodeAgentTokenFilePath = NodeAgentBaseDir + "/token"
+	// NodeAgentTokenSecretName is the name of the secret that contains the shoot access
+	// token of the gardener-node-agent.
+	NodeAgentTokenSecretName = "gardener-node-agent"
+	// NodeAgentTokenSecretKey is the key inside the gardener-node-agent token secret to access
+	// the token.
+	NodeAgentTokenSecretKey = "token"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeAgentConfiguration defines the configuration for the gardener-node-agent.
+type NodeAgentConfiguration struct {
```

Comment:
@rfranzke the NodeAgentConfiguration will never be provided by the operator. It will be generated by gardener and wrapped into a cloud config secret. Does it make sense to provide an example in that case?